clang/test/Analysis/solver-sym-simplification-on-assumption.c

```diff
@@ -0,0 +1,31 @@
+// RUN: %clang_analyze_cc1 %s \
+// RUN:   -analyzer-checker=debug.ExprInspection \
+// RUN:   -verify
+
+void clang_analyzer_eval(int);
+
+void test_derived_sym_simplification_on_assume(int s0, int s1) {
+  int elem = s0 + s1 + 1;
+  if (elem-- == 0) // elem = s0 + s1
+    return;
+
+  if (elem-- == 0) // elem = s0 + s1 - 1
+    return;
+
+  if (s0 < 1) // s0: [1, 2147483647]
```

Comment:
Fixed!

Reply:
Please match against the whole output of the dumps because otherwise if it appears in a substring of the original dump it would be still accepted. By having full dumps one could reduce the chance of having accidental matches.

Reply:
Fixed!